Consolidate to one `schedule` param (#25410)
mistercrunch/Airflow: newsfragments/25410.significant.rst

Deprecation of ``schedule_interval`` and ``timetable`` params

> We add new DAG parameter ``schedule`` that can accept a cron expression, timedelta object, *timetable* object, or list of dataset objects. Params ``schedule_interval`` and ``timetable`` are  deprecated.

Reviewer: Might be worth adding a code example, "this becomes this" type thing.

Author: done

> Deprecation of ``schedule_interval`` and ``timetable`` params

Reviewer 2: ```suggestion
Deprecation of ``schedule_interval`` and ``timetable`` arguments
```

> We add new DAG parameter ``schedule`` that can accept a cron expression, timedelta object, *timetable* object, or list of dataset objects. Params ``schedule_interval`` and ``timetable`` are  deprecated.

Reviewer 2: ```suggestion
We added new DAG argument ``schedule`` that can accept a cron expression, timedelta object, *timetable* object, or list of dataset objects. Arguments ``schedule_interval`` and ``timetable`` are deprecated.
```

> So if you previously used the ``@daily`` cron preset, your dag may have looked like this:

Reviewer 2: ```suggestion
If you previously used the ``@daily`` cron preset, your DAG may have looked like this:
```

.. code-block:: python
    with DAG(
        dag_id='my_example',
        start_date=datetime(2021, 1, 1),
        schedule_interval='@daily',
    ) as dag:
        ...

> Going forward, instead you should use the ``schedule`` param instead:

Reviewer 2: ```suggestion
Going forward, you should use the ``schedule`` argument instead:
```

.. code-block:: python
    with DAG(
        dag_id='my_example',
        start_date=datetime(2021, 1, 1),
        schedule='@daily',
    ) as dag:
        ...

> And the same is true if you used a custom timetable.  Previously you would have used the ``timetable`` param:

Reviewer 2: ```suggestion
The same is true if you used a custom timetable.  Previously you would have used the ``timetable`` argument:
```

.. code-block:: python
    with DAG(
        dag_id='my_example',
        start_date=datetime(2021, 1, 1),
        timetable=EventsTimetable(event_dates=[pendulum.datetime(2022, 4, 5)]),
    ) as dag:
        ...

> Now you should use the ``schedule`` param:

Reviewer 2: ```suggestion
Now you should use the ``schedule`` argument:
```

.. code-block:: python
    with DAG(
        dag_id='my_example',
        start_date=datetime(2021, 1, 1),

>         timetable=EventsTimetable(event_dates=[pendulum.datetime(2022, 4, 5)]),

Reviewer: ```suggestion
        schedule=EventsTimetable(event_dates=[pendulum.datetime(2022, 4, 5)]),
```
